### ISSUE TITLE
Make @Traced inherited

### ISSUE THREAD

**Issue by ggam** [eclipse/microprofile-opentracing#188]
Excuse me if this has been already discussed.

I usually create a CDI stereotype which ressembles the `@Stateless` EJB annotation and can be used on all business classes:
```
@Stereotype
@Retention(RUNTIME)
@Target(TYPE)
@RequestScoped
public @interface Service {
}
```

I'd like to add the `@Traced` annotation to it in order to get automatic OpenTracing support, but sadly the annotations isn't inherited, so it doesn't work.

Did you have any reason not to make `@Traced` inheritable? I can provide a PR for it if you think it's a good idea. In the meantime, I'll simulate the behaviour with a CDI extension adding the annotation to the all beans that use my stereotype.

**Comment by pavolloffay** [comment]
This hasn't been discussed. I am not familiar with this concept. @pilhuhn  or @fmhwong could you please loop in?

**Comment by fmhwong** [comment]
Did you mean adding `@Traced` in the following 

```
@Stereotype
@Retention(RUNTIME)
@Target(TYPE)
@RequestScoped
@Traced
public @interface Service {
}


public class ThingOne {

    @Service
    public void someService() {
        // do something
    } 

}
```

and `someService()` is not traced.

**Comment by ggam** [comment]
Exactly. Since @Traced is not an @Inherited annotation, the target class
doesn't contain it at runtime, and so the interceptor doesn't apply.
